### PR TITLE
remove docs from README Template and simplify Documentation section

### PR DESCRIPTION
**Pull request by MaxUlysse**
The link to the folder docs in the Documentation section is not working:

https://nf-co.re/rnaseq/dev#documentation
-> https://nf-co.re/rnaseq/dev/docs

I propose to actually remove the mention of this folder, and link directly the md files which should simplify the whole section.

Only issue, I would want to link `docs/usage.md` and `docs/output.md` which are not accessible in the current website from this README.

Any idea on how to improve all that?

## PR checklist

 - [ ] This comment contains a description of changes (with reason)
 - [ ] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

**Comment by MaxUlysse** [comment]
Or maybe we can rethink the documentation section in this README, since we're linking the usage already in the section just before. (which I still think should be docs/usage anyway)

**Comment by mashehu** [comment]
I agree with the removal of the mention of the folder in this section. 
The source markdown files are anyway linked at the top of the page (via the edit buttom) or the very bottom with a direct link. 

I would not link to the usage and output markdown files in the readme, since I guess 99% of the people reading it, don't have any intention of editing it or reading the raw markdown file. The rest should either be able to find the button/link or go through the normal contribution path.

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ See [usage docs](https://nf-co.re/{{ cookiecutter.short_name }}/usage) for all o
 
 ## Documentation
 
-The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline which you can read at [https://nf-co.re/{{ cookiecutter.short_name }}/usage](https://nf-co.re/{{ cookiecutter.short_name }}/usage) and [https://nf-co.re/{{ cookiecutter.short_name }}/output](https://nf-co.re/{{ cookiecutter.short_name }}/output) or find in the [`docs/` directory](docs) in the repository.
+The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline: [docs/usage](docs/usage.md) and [docs/output](docs/output.md).
```
Comment (mashehu) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:43]:
```suggestion
The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline: [usage](https://nf-co.re/{{ cookiecutter.short_name }}/usage.md) and [output](https://nf-co.re/{{ cookiecutter.short_name }}/output.md).
```
I would put the full link for portability reasons.

Comment (ewels) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:43]: If the full link, can drop the `.md` too..

Comment (ewels) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:43]:
```suggestion
The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline: [usage](https://nf-co.re/{{ cookiecutter.short_name }}/usage) and [output](https://nf-co.re/{{ cookiecutter.short_name }}/output).
```

**Comment by ewels** [comment]
Along these lines, it could be kind of nice to add something to the markdown files explaining that they should be viewed on the nf-core website because they are missing auto-generated content.

Any thoughts on this? Have a top section followed by a `---` that is then trimmed off on the website or something?

**Comment by ewels** [comment]
x-ref https://github.com/nf-core/tools/issues/727

**Comment by ewels** [comment]
Maybe move https://github.com/nf-core/tools/pull/735#issuecomment-692015883 into a new issue 🚀

**Comment by mashehu** [comment]
> Maybe move [#735 (comment)](https://github.com/nf-core/tools/pull/735#issuecomment-692015883) into a new issue 🚀

Isn't that #727?

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ See [usage docs](https://nf-co.re/{{ cookiecutter.short_name }}/usage) for all o
 
 ## Documentation
 
-The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline which you can read at [https://nf-co.re/{{ cookiecutter.short_name }}/usage](https://nf-co.re/{{ cookiecutter.short_name }}/usage) and [https://nf-co.re/{{ cookiecutter.short_name }}/output](https://nf-co.re/{{ cookiecutter.short_name }}/output) or find in the [`docs/` directory](docs) in the repository.
+The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline: [usage](https://nf-co.re/{{ cookiecutter.short_name }}/usage) and [output](https://nf-co.re/{{ cookiecutter.short_name }}/output).
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:43]:
```suggestion
The {{ cookiecutter.name }} pipeline comes with documentation about the pipeline: [usage](https://nf-co.re/{{ cookiecutter.short_name }}/usage), [pipeline parameters](https://nf-co.re/{{ cookiecutter.short_name }}/parameters) and [output](https://nf-co.re/{{ cookiecutter.short_name }}/output).
```

**Review by ewels — APPROVED** [review]
LGTM, now with the additional parameters.

**Pull request merged**